### ISSUE TITLE
Sentinel executes templates even when it can't reach etcd

### ISSUE THREAD

**Issue by anoldguy** [BlueDragonX/sentinel#7]
Hi Ryan!

We've noticed our config files getting overwritten when sentinel can't contact the etcd servers.  Specifically, this is biting us at boot time.  Here's what we're seeing.  Our files are being modified right when we reboot our box, and get overwritten with the empty template.

```
root@host:/path/to/app/config# ll
drwxr-xr-x  2 user group 4096 Jan 13 23:30 ./
drwxr-xr-x 13 user group 4096 Nov 28 21:38 ../
-rw-------  1 user group   69 Jan 13 23:30 database_config.yml
-rw-------  1 user group   69 Jan 13 23:30 memcached_config.yml
root@host:/path/to/app/config# cat memcached_config.yml
# This is a sentinel-generated config file!
# Do not edit by hand!

root@host:/path/to/app/config#
```
The last modified time corresponds to the sentinel start times:

```
root@host:/path/to/app/config# ps axuwwf | grep sentinel
user        651  0.0  0.0 399556  4792 ?        Ssl  23:30   0:00 /usr/local/bin/sentinel -config=/etc/sentinel.d/app4_staging.yml
user        656  0.0  0.0 399556  4672 ?        Ssl  23:30   0:00 /usr/local/bin/sentinel -config=/etc/sentinel.d/app3_staging.yml
user        657  0.0  0.0 399684  4668 ?        Ssl  23:30   0:00 /usr/local/bin/sentinel -config=/etc/sentinel.d/app2_staging.yml
user        658  0.0  0.0 399684  4760 ?        Ssl  23:30   0:00 /usr/local/bin/sentinel -config=/etc/sentinel.d/app1_staging.yml
```

We've got our sentinels configured via upstart, and I noticed it is starting sentinel before the network is available.  Fair enough. So we modify our upstart config to wait until the specific interface we need is up and ready.  No dice. It turns out that udev tells upstart the interface is up about 10s before the bond is able to send any traffic.  So we go lo-fi and put a sleep in place. Yuk.  However, it works, and sentinel writes correct configs.

However, I'm proposing that it is incorrect behavior to execute the templates for the watchers if you can't talk to etcd.

In the sentinel logs (during another reboot test), we see:
```
sentinel 2015/01/14 04:40:54.605180 [DEBUG]  get key '/shared/app1/staging/memcached': 501: All the given peers are not reachable (Tried to connect to each peer twice and failed) [0]
sentinel 2015/01/14 04:40:54.605192 [DEBUG]  context: map[]

sentinel 2015/01/14 04:40:54.609909 [DEBUG]  template '/etc/sentinel.d/app1_staging/generic.tpl' rendered to '/path/to/app/config/memcached.yml'
sentinel 2015/01/14 04:40:54.609926 [DEBUG]  app1_memcached has no command, skipping
sentinel 2015/01/14 04:40:54.609932 [INFO]   app1_memcached executed
```
This implied we had an issue with our etcd setup, so we started investigating that, but it turns out tracking down this error message leads me to the [go-etcd package](https://github.com/coreos/go-etcd/blob/master/etcd/error.go#L8-10). So from sentinel's standpoint, it got a 501 from etcd, but still executed the template with an empty context, which, of course, was written to disk, as the result was different.

I do not know golang, but I think if there were a check in the [watcher#Run function](https://github.com/BlueDragonX/sentinel/blob/master/watcher.go#L106-115), that would inspect the event and see if it were empty, or maybe in [client#GetMaps function](https://github.com/BlueDragonX/sentinel/blob/master/client.go#L110-121), detect if the map is empty and return an error. I don't know.

However, I do think it's a bug that sentinel re-renders the templates when it can't contact etcd.  So out of this I think 2 things should happen:

1. sentinel doesn't execute templates when etcd is unavailable
2. sentinel either dies if etcd is unavailable or retries, preferably with a backoff so we don't keep hammering.

**Comment by BlueDragonX** [comment]
I agree and was not aware of this. Coupled with the fact that it is not render when the connection picks up is a big issue. I'll start looking into this today as a think I finally have some cycles between some of the fires I've been putting out!

**Comment by BlueDragonX** [comment]
Fixed in v2.0.0.
